### ISSUE TITLE
bug(COMPONENT): Accessibility for dialog, when open a dialog, the app-root will add aria-hidden='true',and the dialog content still has aria-hdden='true' tabindex=0 element.

### ISSUE THREAD

**Issue by lovertommorrow** [angular/components#28655]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

When use Axe DevTools check dialog, the app-root element will add the aria-hidden="true", dialog content still has the element tabindex="0" aria-hidden="true". and use Axe Devtools check, it will report: Ensures aria-hidden elements are not focusable nor contain focusable elements.

### Reproduction

StackBlitz link: 
Steps to reproduce:
1. Open dialog
2. Use axe Devtools check web page
3.Ensures aria-hidden elements are not focusable nor contain focusable elements

### Expected Behavior

if set the aria-hdden="true", this element should can not be focusable. 

### Actual Behavior

if set the aria-hdden="true", this element should can be focusable. 

### Environment

- Angular:17.1.2
- CDK/Material:17.1.2
- Browser(s):121.0.6167.189
- Operating System (e.g. Windows, macOS, Ubuntu):Windows
